scripts/k6/stream-api/main.js

```diff
@@ -0,0 +1,166 @@
+import {check} from 'k6';
+import http from "k6/http";
+import { Counter } from 'k6/metrics';
+import {randomItem} from 'https://jslib.k6.io/k6-utils/1.2.0/index.js';
+import { URL } from 'https://jslib.k6.io/url/1.0.0/index.js';
+import {Api, randomPayloads} from "./api.js";
+
+const SCENARIO = __ENV.K6_SCENARIO || "constant"; // constant or ramping
+const TABLE_MAPPING = __ENV.K6_TABLE_MAPPING || "static"; // static or template
+
+// Target
+const API_TOKEN = __ENV.K6_API_TOKEN;
+const API_HOST = __ENV.K6_API_HOST || "http://localhost:8001";
+const OVERWRITE_SOURCE_HOST = __ENV.K6_OVERWRITE_SOURCE_HOST; // to call directly k8s service
+
+// Common for all scenarios
+const PARALLEL_REQS_PER_USER = __ENV.K6_PARALLEL_REQS_PER_USER || 1;
+
+// Constant VUs / iterations scenario
+const CONST_VIRTUAL_USERS = __ENV.K6_CONST_VIRTUAL_USERS || 1000;
+const CONST_TOTAL_REQUESTS = __ENV.K6_CONST_TOTAL_REQUESTS || 1000000;
+const CONST_TIMEOUT = __ENV.K6_CONST_TIMEOUT || "20m";
+
+// Ramping scenario
+const RAMPING_MAX_VIRTUAL_USERS = __ENV.K6_RAMPING_MAX_VIRTUAL_USERS || 1000;
+const RAMPING_UP_DURATION = __ENV.K6_RAMPING_UP_DURATION || "2m";
+const RAMPING_STABLE_DURATION = __ENV.K6_RAMPING_STABLE_DURATION || "10m";
+const RAMPING_DOWN_DURATION = __ENV.K6_RAMPING_DOWN_DURATION || "2m";
+
+// Stream configuration
+const SYNC_MODE = __ENV.STREAM_SYNC_MODE || "disk"; // cache / disk
+const SYNC_WAIT = __ENV.STREAM_SYNC_WAIT || "1"; // 1 = enabled, 0 = disabled
+
+// Available scenarios, use K6_SCENARIO env to select one
+const scenarios = {
+  constant: {
+    executor: "shared-iterations",
+    vus: CONST_VIRTUAL_USERS,
+    iterations: CONST_TOTAL_REQUESTS / PARALLEL_REQS_PER_USER,
+    maxDuration: CONST_TIMEOUT,
+  },
+  ramping: {
+    executor: 'ramping-vus',
+    startVUs: 0,
+    stages: [
+      {target: RAMPING_MAX_VIRTUAL_USERS, duration: RAMPING_UP_DURATION},
+      {target: RAMPING_MAX_VIRTUAL_USERS, duration: RAMPING_STABLE_DURATION},
+      {target: 0, duration: RAMPING_DOWN_DURATION},
+    ],
+  }
+}
+
+const mappings = {
+  static: {
+    columns: [
+      {type: "uuid", name: "id", primaryKey: true},
+      {type: "datetime", name: "datetime"},
+      {type: "ip", name: "ip"},
+      {type: "body", name: "body"},
+      {type: "headers", name: "headers"},
+    ],
+  },
+  template: {
+    columns: [
+      {type: "uuid", name: "id", primaryKey: true},
+      {
+        type: "template",
+        name: "template",
+        template: {
+          language: "jsonnet",
+          undefinedValueStrategy: "null",
+          content: `Body('a')+":"+Body('c.f.g')`,
+        },
+      },
+    ],
+  }
+}
```

Comment:
I removed the  static.js, template.js - they were same - replaced by the  K6_TABLE_MAPPING env.